applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_static_solver.py

```diff
@@ -198,18 +198,38 @@ def _create_linear_solver(self):
                 KM.Logger.PrintWarning("Using MixedULMLinearSolver, definition of ALM parameters recommended")
                 name_mixed_solver = self.contact_settings["mixed_ulm_solver_parameters"]["solver_type"].GetString()
                 if (name_mixed_solver == "MixedULMLinearSolver"):
+                    linear_solver_name = self.settings["linear_solver_settings"]["solver_type"].GetString()
```

Comment:
why this? it should detect this internally

Reply:
The current linear solver factory lacks that capability, we should push #615